File: backend/src/main/java/com/core/entity/Exam.java
package com.core.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "exams")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Exam extends BaseEntity {

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    private String subject;

    @Column(name = "total_points")
    private Integer totalPoints;

    @Column(name = "time_limit") // in minutes
    private Integer timeLimit;

    @Column(name = "passing_score")
    private Integer passingScore;

    @Column(name = "published_at")
    private LocalDateTime publishedAt;

    @Enumerated(EnumType.STRING)
    private ExamStatus status = ExamStatus.DRAFT;

    @Column(name = "shuffle_questions")
    private Boolean shuffleQuestions = false;

    @OneToMany(mappedBy = "exam", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Question> questions;

    public enum ExamStatus {
        DRAFT, SCHEDULED, ACTIVE, ARCHIVED
    }
}

File: frontend/src/types/exam.ts
export interface ExamResponse {
  id: number;
  title: string;
  description: string;
  subject: string;
  totalPoints: number;
  timeLimit: number;
  passingScore: number;
  publishedAt: string;
  status: 'DRAFT' | 'SCHEDULED' | 'ACTIVE' | 'ARCHIVED';
  shuffleQuestions: boolean;
  creatorName: string;
  createdAt: string;
  updatedAt: string;
}

export interface ExamRequest {
  title: string;
  description?: string;
  subject?: string;
  totalPoints: number;
  timeLimit?: number;
  passingScore?: number;
  publishedAt?: string;
  shuffleQuestions?: boolean;
}